src/test/ui/print_type_sizes/generator_discr_placement.rs

```diff
@@ -0,0 +1,25 @@
+// compile-flags: -Z print-type-sizes
+// build-pass
+// ignore-pass
+
+// Tests a generator that has its discriminant as the *final* field.
+
+// Avoid emitting panic handlers, like the rest of these tests...
```

Comment:
If the other tests do this already as it says it's fine but this should probably be a lib crate instead.